File: liqp-core/src/main/java/liqp/nodes/ContainsNode.java
package liqp.nodes;

import java.util.Arrays;
import liqp.node.LNode;
import lombok.Getter;

@Getter
public class ContainsNode extends ExpressionNode {

    public ContainsNode(LNode lhs, LNode rhs) {
      super(lhs, rhs);
    }

    @Override
    public Object render(LContext context) {

        Object collection = lhs.render(context);
        Object needle = rhs.render(context);

        if(Companion.isArray(collection)) {
            Object[] array = Companion.asArray(collection);
            return Arrays.asList(array).contains(needle);
        }

        if(Companion.isString(collection)) {
            return Companion.asString(collection).contains(Companion.asString(needle));
        }

        return false;
    }
}

File: liqp-core/src/main/kotlin/liqp/nodes/ContainsNode.kt
package liqp.nodes

import com.google.common.collect.ImmutableSet
import liqp.context.LContext
import liqp.node.LNode

class ContainsNode(lhs: LNode, rhs: LNode) : ExpressionNode(lhs, rhs) {

  override fun render(context: LContext): Any? {

    val collection = lhs.render(context)
    val needle = rhs.render(context)

    if (context.isIterable(collection)) {
      val array = context.asIterable(collection)
      return ImmutableSet.copyOf(array).contains(needle)
    }

    val string = context.asString(collection)
    val needlStr = context.asString(needle)
    return when (needlStr) {
      null -> false
      else -> string?.contains(needlStr) ?: false
    }
  }
}
